app/src/main/kotlin/org/cru/godtools/ui/dashboard/optinnotification/OptInNotificationModalOverlay.kt

```diff
@@ -0,0 +1,230 @@
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.slack.circuit.overlay.Overlay
+import com.slack.circuit.overlay.OverlayNavigator
+import kotlinx.coroutines.launch
+import org.ccci.gto.android.common.androidx.compose.foundation.layout.padding
+import org.cru.godtools.R
+import org.cru.godtools.util.isTablet
+
+enum class PermissionStatus {
+    APPROVED, // Approved
+    SOFT_DENIED,  // Denied but requestable
+    HARD_DENIED, // Denied and no longer requestable
+    UNDETERMINED // First time request
+}
+
+class OptInNotificationModalOverlay(
+    val requestPermission: suspend () -> Unit,
+    val isHardDenied: Boolean,
+) : Overlay<Unit> {
+
+    @Composable
+    override fun Content(navigator: OverlayNavigator<Unit>) {
+        BackHandler { navigator.finish(Unit) }
+
+        val isTablet = isTablet()
+
+        val coroutineScope = rememberCoroutineScope()
+        val transitionState = remember { MutableTransitionState(false) }
+
+        LaunchedEffect(Unit) { transitionState.targetState = true }
+
+        LaunchedEffect(transitionState.currentState, transitionState.targetState) {
+            if (!transitionState.currentState && transitionState.isIdle) {
+                navigator.finish(Unit)
+            }
+        }
+
+        Surface(
+            color = Color.Transparent, modifier = Modifier
+                .fillMaxSize()
+                .background(Color.Black.copy(alpha = 0.2f))
+        ) {
+            AnimatedVisibility(
+                visibleState = transitionState,
+                enter = slideInVertically(
+                    initialOffsetY = { it },
+                    animationSpec = tween(
+                        durationMillis = 350,
+                    ),
+                ),
+                exit = slideOutVertically(
+                    targetOffsetY = { it },
+                    animationSpec = tween(
+                        durationMillis = 350,
+                    ),
+                ),
+            ) {
+                BoxWithConstraints {
+                    val totalWidth = constraints.maxWidth.dp
+                    val isLargeTablet = totalWidth > 1700.dp
+
+                    Card(
+                        modifier = Modifier
+                            .padding(horizontal = if (isTablet) if (isLargeTablet) 200.dp else 150.dp else 16.dp)
```

Comment:
this is confusing, let's convert this to a when statement to be more clear.

```suggestion
                            .padding(
                                horizontal = when {
                                    isLargeTablet -> 200.dp
                                    isTablet -> 150.dp
                                    else -> 16.dp
                                }
                            )
```